RTD: update the build-from-CIAO documentation

DougBurke opened the pull request:
# Summary

Update the "build with CIAO" documentation to match the CIAO 4.14 conda instructions and automate the process.

# Details

Adds the `scripts/use_ciao_config` script that automates the setting of the `setup.cfg` file for users of CIAO installed with conda, and update the documentation to match.

Note that this is how I found issue #1452 and changes to that could influence this, but they can be handled separately.

wmclaugh commented on docs/developer/index.rst:
```diff
@@ -129,50 +136,48 @@ As described in :ref:`build-from-source`, the file ``setup.cfg`` in
-lists the lines in ``setup.cfg`` that need to be modified (adjust
-xspec version as needed)::
+variable ``ASCDS_INSTALL`` (or ``CONDA_PREFIX``, which has the same
+value) can be used. For most cases, the ``scripts/use_ciao_cofig``
```
```suggestion
value) can be used. For most cases, the ``scripts/use_ciao_config``
```

wmclaugh commented on docs/developer/index.rst:
```diff
@@ -129,50 +136,48 @@ As described in :ref:`build-from-source`, the file ``setup.cfg`` in
+
+then change the ``setup.cfg`` to change the following lines, noting
+that the `${ASCDS_INSTALL}` environment variable **must** be
+replaced by it's actual value, and the ``xspec_version`` line
```
```suggestion
replaced by its actual value, and the ``xspec_version`` line
```

wmclaugh commented on scripts/use_ciao_config:
```diff
@@ -0,0 +1,99 @@
+fi
+
+# Extract the XSPEC version from the conda packaging metadata;
+# hopefully the formatting and structure remani consistent.
```
```suggestion
# hopefully the formatting and structure remain consistent.
```

hamogu reviewed (commented):
This looks good to me (I have not tried the script), but it might be useful to get CM's take since they make the conda pacakges and know best what is installed where and when.

DougBurke commented:
@Marie-Terrell or @harlantc - can I add this onto your plate of things to do, given the comment above https://github.com/sherpa/sherpa/pull/1455#pullrequestreview-930429700

DougBurke commented:
I've rebased this (to pick up some unrelated doc-build changes) and to address comments made by @Marie-Terrell . As I was rebasing I just squashed everything into one commit.

DougBurke commented:
I've rebased as #1484 has been merged. I need to update the script to support this - following https://github.com/sherpa/sherpa/pull/1484#issuecomment-1111327433 and to add a documentation note to point out this doesn't quite match the CIAO build process

DougBurke commented:
I've updated this to take advantage of #1484 and to add the warning that "this doesn't quite match CIAO" to the documentation (replacing the warning I had added to say we couldn't support #1484, as that is now wrong ;-)

wmclaugh commented on docs/developer/index.rst:
```diff
@@ -107,72 +107,101 @@ How do I ...
+    conda install -n sherpaciao --only-deps -c https://cxc.cfa.harvard.edu/conda/ciao -c conda-forge sherpa
+    conda activate sherpaciao
 
 The first line installes the full `CIAO release
```
Not changed as part of this PR but  may as well correct it.

DougBurke commented on scripts/use_ciao_config:
```diff
@@ -0,0 +1,104 @@
+sed -i "s|#with-xspec=True|with-xspec=True|" $cfg
+sed -i "s|#xspec_include_dirs = None|xspec_include_dirs=${CONDA_PREFIX}/include|g" $cfg
+sed -i "s|#xspec_lib_dirs = None|xspec_lib_dirs=${CONDA_PREFIX}/lib|g" $cfg
+sed -i "s|#xspec_version = .*|xspec_version = ${xspec_version}|g" $cfg
```
An accident - I'll remove the global flag as it's not needed here.

DougBurke commented on scripts/use_ciao_config:
```diff
@@ -0,0 +1,104 @@
+sed -i "s|#fftw-libraries=fftw3|fftw-libraries=fftw3|" $cfg
+
+sed -i "s|#with-xspec=True|with-xspec=True|" $cfg
+sed -i "s|#xspec_include_dirs = None|xspec_include_dirs=${CONDA_PREFIX}/include|g" $cfg
```
```suggestion
sed -i "s|#xspec_include_dirs = None|xspec_include_dirs=${CONDA_PREFIX}/include|" $cfg
```

DougBurke commented on scripts/use_ciao_config:
```diff
@@ -0,0 +1,104 @@
+
+sed -i "s|#with-xspec=True|with-xspec=True|" $cfg
+sed -i "s|#xspec_include_dirs = None|xspec_include_dirs=${CONDA_PREFIX}/include|g" $cfg
+sed -i "s|#xspec_lib_dirs = None|xspec_lib_dirs=${CONDA_PREFIX}/lib|g" $cfg
```
```suggestion
sed -i "s|#xspec_lib_dirs = None|xspec_lib_dirs=${CONDA_PREFIX}/lib|" $cfg
```

DougBurke commented on scripts/use_ciao_config:
```diff
@@ -0,0 +1,104 @@
+sed -i "s|#with-xspec=True|with-xspec=True|" $cfg
+sed -i "s|#xspec_include_dirs = None|xspec_include_dirs=${CONDA_PREFIX}/include|g" $cfg
+sed -i "s|#xspec_lib_dirs = None|xspec_lib_dirs=${CONDA_PREFIX}/lib|g" $cfg
+sed -i "s|#xspec_version = .*|xspec_version = ${xspec_version}|g" $cfg
```
```suggestion
sed -i "s|#xspec_version = .*|xspec_version = ${xspec_version}|" $cfg
```

hamogu replied:
Just a comment: In  the GH UI on the web, you can mark more than one line with the mouse (click on the + next to the line numbers, hold mouse button and pull). When you then select "Add a suggestion" you can edit multiple lines at one so you don't need to do it for each line separately.
